cmd/dockerd/config_common_unix.go

```diff
@@ -15,6 +20,24 @@ var (
 	defaultExecRoot = "/var/run/docker"
 )
 
+func init() {
+	if !rootless.RunningWithNonRootUsername {
+		return
+	}
+	//  pam_systemd sets XDG_RUNTIME_DIR but not other dirs.
+	if xdgDataHome := os.Getenv("XDG_DATA_HOME"); xdgDataHome != "" {
```

Comment:
Wondering if other parts should now also take `XDG_DATA_HOME` into account, otherwise files may be split between `XDG_DATA_HOME` and `HOME`

Reply:
Cert paths (`cmd/dockerd/options.go`, `cli/config/configdir.go`) need to ignore `XDG_DATA_HOME` and use `~/.docker` or `DOCKER_CONFIG`, for consistency with the client implementation.